safehttp/plugins/fetch_metadata/fetch_metadata_test.go

```diff
@@ -0,0 +1,407 @@
+			site:   "same-site",
+		},
+		{
+			name:   "user agent initiated",
+			method: safehttp.MethodGet,
+			site:   "none",
+		},
+		{
+			name:   "cors bug missing mode",
+			method: safehttp.MethodOptions,
+			site:   "cross-site",
+			mode:   "",
+		},
+	}
+
+	allowedRIPNavHeaders = []testHeaders{
+		{
+			name:   "cross origin GET navigate from document",
+			method: safehttp.MethodGet,
+			site:   "cross-site",
+			mode:   "navigate",
+			dest:   "document",
+		},
+		{
+			name:   "cross origin HEAD navigate from document",
+			method: safehttp.MethodHead,
+			site:   "cross-site",
+			mode:   "navigate",
+			dest:   "document",
+		},
+		{
+			name:   "cross origin GET navigate from nested-document",
+			method: safehttp.MethodGet,
+			site:   "cross-site",
+			mode:   "navigate",
+			dest:   "nested-document",
+		},
+		{
+			name:   "cross origin HEAD navigate from nested-document",
+			method: safehttp.MethodHead,
+			site:   "cross-site",
+			mode:   "navigate",
+			dest:   "nested-document",
+		},
+		{
+			name:   "cross origin GET nested-navigate from document",
+			method: safehttp.MethodGet,
+			site:   "cross-site",
+			mode:   "nested-navigate",
+			dest:   "document",
+		},
+		{
+			name:   "cross origin HEAD nested-navigate from document",
+			method: safehttp.MethodHead,
+			site:   "cross-site",
+			mode:   "nested-navigate",
+			dest:   "document",
+		},
+		{
+			name:   "cross origin GET nested-navigate from nested-document",
+			method: safehttp.MethodGet,
+			site:   "cross-site",
+			mode:   "nested-navigate",
+			dest:   "nested-document",
+		},
+		{
+			name:   "cross origin HEAD nested-navigate from nested-document",
+			method: safehttp.MethodHead,
+			site:   "cross-site",
+			mode:   "nested-navigate",
+			dest:   "nested-document",
+		},
+	}
+
+	disallowedRIPNavHeaders = []testHeaders{
+		{
+			name:   "cross origin POST",
+			method: safehttp.MethodPost,
+			site:   "cross-site",
+			mode:   "navigate",
+			dest:   "document",
+		},
+		{
+			name:   "cross origin GET from object",
+			method: safehttp.MethodGet,
+			site:   "cross-site",
+			mode:   "navigate",
+			dest:   "object",
+		},
+		{
+			name:   "cross origin HEAD from embed",
+			method: safehttp.MethodHead,
+			site:   "cross-site",
+			mode:   "navigate",
+			dest:   "embed",
+		},
+	}
+
+	disallowedRIPHeaders = []testHeaders{
+		{
+			name:   "cross origin no cors",
+			method: safehttp.MethodPost,
+			site:   "cross-site",
+			mode:   "cors",
+			dest:   "document",
+		},
+		{
+			name:   "cross origin no cors",
+			method: safehttp.MethodPost,
+			site:   "cross-site",
+			mode:   "no-cors",
+			dest:   "nested-document",
+		},
+	}
+)
+
+func TestAllowedResourceIsolationEnforceMode(t *testing.T) {
+	tests := allowedRIPHeaders
+	for _, test := range allowedRIPNavHeaders {
+		tests = append(tests, testHeaders{
+			name:   test.name,
+			method: test.method,
+			site:   test.site,
+			mode:   test.mode,
+			dest:   test.dest,
+		})
+	}
```

Comment:
You can append two slices using `append` like this:
```go
x = append(slice1, slice2...)
```

Reply:
Oh, nice, thanks :)